packages/contract/src/contract-factory.ts

```diff
@@ -146,26 +154,175 @@ export default class ContractFactory {
+  }
+
+  /**
+   * Deploy a contract with the specified options.
+   *
+   * @param deployOptions - Options for deploying the contract.
+   * @returns A promise that resolves to the deployed contract instance.
+   */
+  async deployContract<TContract extends Contract = Contract>(
+    deployOptions: DeployContractOptions = {}
+  ): Promise<DeployContractResult<TContract>> {
+    const account = this.getAccount();
+    const { consensusParameters } = account.provider.getChain();
+    const maxContractSize = consensusParameters.contractParameters.contractMaxSize.toNumber();
+
+    if (this.bytecode.length > maxContractSize) {
       throw new FuelError(
         ErrorCode.CONTRACT_SIZE_EXCEEDS_LIMIT,
-        'Contract bytecode is too large. Max contract size is 100KB'
+        'Contract bytecode is too large. Max contract size is 100KB' // change error
       );
     }
 
     const { contractId, transactionRequest } = await this.prepareDeploy(deployOptions);
 
+    const transactionResponse = await account.sendTransaction(transactionRequest);
+
+    const waitForResult = async () => {
+      const transactionResult = await transactionResponse.waitForResult<TransactionType.Create>();
+      const contract = new Contract(contractId, this.interface, account) as TContract;
+
+      return { contract, transactionResult };
+    };
+
+    return { waitForResult, contractId, transactionId: transactionResponse.id };
+  }
+
+  /**
+   * Chunks and deploys a contract via a loader contract. Suitable for deploying contracts larger than the max contract size.
+   *
+   * @param deployOptions - Options for deploying the contract.
+   * @returns A promise that resolves to the deployed contract instance.
+   */
+  async deployContractAsBlobs<TContract extends Contract = Contract>(
+    deployOptions: DeployContractOptions = {
+      chunkSizeTolerance: CHUNK_SIZE_TOLERANCE,
+    }
+  ): Promise<DeployContractResult<TContract>> {
     const account = this.getAccount();
+    const { configurableConstants, chunkSizeTolerance } = deployOptions;
+    if (configurableConstants) {
+      this.setConfigurableConstants(configurableConstants);
+    }
 
-    const transactionResponse = await account.sendTransaction(transactionRequest);
+    // Generate the chunks based on the maximum chunk size
+    const chunkSize = this.getMaxChunkSize(chunkSizeTolerance);
+    const chunks = getContractChunks(this.bytecode, chunkSize).map((c) => {
+      const transactionRequest = this.blobTransactionRequest({ bytecode: c.bytecode });
+      return {
+        ...c,
+        transactionRequest,
+        blobId: transactionRequest.blobId,
+      };
+    });
+
+    // Check the account can afford to deploy all chunks
+    const costs = [];
+    for (const { transactionRequest } of chunks) {
+      const { minFee } = await account.getTransactionCost(transactionRequest);
+      costs.push(minFee);
+    }
+    const totalCost = costs.reduce((acc, cost) => acc.add(cost), bn(0));
```

Comment:
@danielbate we can calculate the `minFee` directly here without calling `getTransactionCost` since we do not need to dry run it in order to get the gas used. 

I've also filtered the already estimated blob IDs just like already happens when the blob TXs are being submitted. Let me know if that makes sense. 

By doing this we could get a value close between the estimated and real fee values:

```terminal
total estimated fee: 557841

total payed in fee:  559354
```

```suggestion
    let totalCost = bn(0);
    const chainInfo = account.provider.getChain();
    const gasPrice = await account.provider.estimateGasPrice(10);
    const estimatedBlobIds: string[] = [];

    for (const { transactionRequest, blobId } of chunks) {
      if (!estimatedBlobIds.includes(blobId)) {
        const minGas = transactionRequest.calculateMinGas(chainInfo);
        const minFee = calculateGasFee({
          gasPrice,
          gas: minGas,
          priceFactor: chainInfo.consensusParameters.feeParameters.gasPriceFactor,
          tip: transactionRequest.tip,
        }).add(1);

        totalCost = totalCost.add(minFee);
        estimatedBlobIds.push(blobId);
      }
    }
```

Reply:
Great suggestion mate, and good point on filtering the already estimated blobId, makes sense.